lib/factory_bot/strategy/stub.rb

```diff
@@ -79,6 +83,11 @@ def has_settable_id?(result_instance)
           result_instance.class.primary_key
       end
 
+      def uuid_primary_key?(result_instance)
+        result_instance.respond_to?(:column_for_attribute) &&
+          result_instance.column_for_attribute(result_instance.class.primary_key).sql_type == "uuid"
```

Comment:
How do we feel about adding some sort of guard here to protect us from loosing access to a private api here. Could we add a comment noting it? This feels like something future us would appreciate 😄 I could also add that the `column_for_attribute` could return nil(probably not for the primary key, but the API documents this return value), how do we feel about safe navigating after the method call to respect the spec?

Reply:
> How do we feel about adding some sort of guard here to protect us from loosing access to a private api here. Could we add a comment noting it? This feels like something future us would appreciate 😄

When I created the PR, I didn't notice that `column_for_attribute` was private API. Maybe this PR is a bit risky then, but I'm not sure there is an easy workaround. As composerinteralia said: "It's sort of awkward that column_for_attribute is public and returns a connection adapter column, but then none of the methods on the column are public. What good is the method if you can't do anything with the return value?". 

Maybe it's worth asking the Rails team? What kind of guard did you have in mind here?

> I could also add that the column_for_attribute could return nil(probably not for the primary key, but the API documents this return value), how do we feel about safe navigating after the method call to respect the spec?

I'm not sure `column_for_attribute` can return nil here. According to [the method definition](https://github.com/rails/rails/blob/581aac754c6cbe9f3f4de396bc8ef3728e56f4f4/activerecord/lib/active_record/model_schema.rb#L460), in case the column does not exist, it would return a `NullColumn` that would still respond to `sql_type` and return nil. I'm not an ActiveRecord expert, so maybe I'm missing something here 😄.

Reply:
[`column_for_attribute`](https://api.rubyonrails.org/classes/ActiveRecord/ModelSchema/ClassMethods.html#method-i-column_for_attribute) is publicly documented. It's only the `sql_type` method that isn't.

> in case the column does not exist, it would return a NullColumn that would still respond to sql_type and return nil

Yup, that sounds right. `sql_type` could return `nil`, in which case `nil == :uuid` would return false and we'd assume it's not a uuid. But I don't think we'd ever get that, since the stub factory checks that the model has a primary key before getting into this code.